swap_meet/vendor.py

```diff
@@ -1,2 +1,121 @@
+from swap_meet.item import Item
+
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        
+        if inventory == None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+            
+        
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if not item:
+            return False
+        elif item not in self.inventory:
+            return False    
+        
+        self.inventory.remove(item)
+        return item
+        
+    def get_by_category(self, category):
+        new_list = []
+        
+        for item in self.inventory:
+            if category == item.category:
+                new_list.append(item)
+        return new_list
+    
+# ***** added for wave_03 *****
+    def swap_items(self, Vendor, my_item, their_item):
+        
+        if my_item in self.inventory and their_item in Vendor.inventory:
+            # removes item from self inventory and adds it to friend's inventory
+            self.inventory.remove(my_item)
+            Vendor.inventory.append(my_item)
+            
+            # removes item from friend's inventory and adds it to self inventory
+            Vendor.inventory.remove(their_item)
+            self.inventory.append(their_item)
+            
+            return True
+        
+        else:
+            return False
```

Comment:
In this case you would pass in the attribute name instead of Vendor. We don't need to use the class Vendor here. Instead we would use a name like `friend` or `other`: 
```
    def swap_items(self, other, my_item, their_item):

        if my_item not in self.inventory or their_item not in other.inventory:
            return False

        #adding and removing to self
        self.remove(my_item)
        self.add(their_item)

        #adding and removing to other
        other.remove(their_item)
        other.add(my_item)

        return True
```

Reply:
You can return the action of swapping out the items without the else because the swap won't happen because of the guard clause:

```
if not their_best_item  or not my_best_item :
      return False

self.swap_items(other, my_best_item, their_best_item)
      return True
```